src/components/Input/QtySelector/QtySelector.js

```diff
@@ -30,52 +31,80 @@ class QtySelector extends Component {
   static propTypes = {
     disabled: PropTypes.bool,
     value: PropTypes.number,
-    style: PropTypes.objectOf(PropTypes.string)
+    style: PropTypes.objectOf(PropTypes.string),
+    min: PropTypes.number,
+    max: PropTypes.number,
+    onValueUpdated: PropTypes.func
   };
 
   static defaultProps = {
     disabled: false,
     style: {},
-    value: null
+    value: 0,
+    min: QtySelector.MIN_INPUT_VALUE,
+    max: QtySelector.MAX_INPUT_VALUE,
+    onValueUpdated: () => {}
   };
 
   state = {
-    value: this.props.value || "",
+    value: this.props.value,
     focused: false
   };
 
   handleChange = e => {
     const { value } = e.target;
+    const { max, min } = this.props;
 
     if (
       QtySelector.isNumber(value) &&
-      value.length <= QtySelector.MAX_LENGTH_VAL
+      value.length <= QtySelector.MAX_LENGTH_VAL &&
+      value <= max &&
+      value >= min
     ) {
-      this.setState(() => ({ value: Number.parseInt(value, 10) }));
+      this.setState(
+        () => ({ value: Number.parseInt(value, 10) }),
+        () => this.handleValueUpdated()
+      );
     } else if (value === "") {
       // delete input value case
-      this.setState(() => ({ value: "" }));
+      this.setState(() => ({ value: "" }), () => this.handleValueUpdated());
     }
   };
 
-  increment = () =>
-    this.setState(state => {
-      if (state.value === "") {
-        return { value: 0 };
-      }
-      if (state.value === QtySelector.MAX_INPUT_VALUE) {
-        return { value: QtySelector.MAX_INPUT_VALUE };
-      }
-      return { value: state.value + 1 };
-    });
-
-  decrement = () =>
-    this.setState(state => {
-      if (state.value === "" || state.value === 0) {
-        return { value: 0 };
-      }
-      return { value: state.value - 1 };
-    });
+  increment = () => {
+    const { max } = this.props;
+    this.setState(
+      state => {
+        if (state.value === "") {
+          return { value: 0 };
+        }
+        if (state.value === max) {
+          return { value: max };
+        }
+        return { value: state.value + 1 };
+      },
+      () => this.handleValueUpdated()
+    );
+  };
+
+  decrement = () => {
+    const { min } = this.props;
+    this.setState(
+      state => {
+        if (state.value === "" || state.value === min) {
+          return { value: min };
+        }
+        return { value: state.value - 1 };
+      },
+      () => this.handleValueUpdated()
+    );
+  };
+
+  handleValueUpdated = () => {
+    const { onValueUpdated } = this.props;
+    const { value } = this.state;
+    onValueUpdated(value);
```

Comment:
Maybe we should reuse the `onChange` prop here, so we don't have two different callbacks for handling `QtySelector` value updates? Unfortunately, the function signature is a bit different (`onChange` receives an event object) and wouldn't be backwards compatible 🤔

Reply:
I wanted to use the `onChange` prop at first but then I saw it takes an `event` object as its argument. So I had to make a separate prop for it. Do you think if we should still use the onChange prop and add support for different signature?

Reply:
I'm not sure... I guess we should avoid parametric polymorphism, if possible, so the methods can be inlined by the compiler

Reply:
I agree. IMO, I think having `handleValueUpdated` method to cover both scenarios (clicking on buttons and manually editing by keyboard) makes sense. Trying to reuse the `onChange` prop is a little complicated to make it support different type of argument. I'm open to any suggestion to make this better.

Reply:
I agree with you. Please update the documentation for the `QtySelector` component within the `catalog` and then this PR will be ready to be merged.

Reply:
Thank you! I've updated the docs in this commit: https://github.com/ticketmaster/aurora/pull/581/commits/235518d301a541e6e56096957ea0642176377964

In addition, in the above commit, I added a change to re-enable support for the `disabled` prop by bypassing the min and max value condition.

Also, I made a mistake on the `[...Array(max - min + 1)]`, it should be `[...Array(max + 1)]` to support when the min value is larger than 0. I know it's only better than the current `[...Array(1000)]` only when we have a lower max value prop. I really wanted to make it better but several attempts have failed and I have to move on to other tasks 😞

Reply:
👍 Good to go! Thank you